pkg/secrets/keyvault/client.go

```diff
@@ -307,55 +307,112 @@ func (k *KeyvaultSecretClient) Delete(ctx context.Context, key types.NamespacedN
 	return err
 }
 
-// Get gets a key from KeyVault
-func (k *KeyvaultSecretClient) Get(ctx context.Context, key types.NamespacedName) (map[string][]byte, error) {
-	vaultBaseURL := getVaultsURL(ctx, k.KeyVaultName)
-	var secretName string
-	if len(key.Namespace) != 0 {
-		secretName = key.Namespace + "-" + key.Name
+// Delete deletes a key in KeyVault
+func (k *SecretClient) Delete(ctx context.Context, key secrets.SecretKey, opts ...secrets.SecretOption) error {
+	options := &secrets.Options{}
+	for _, opt := range opts {
+		opt(options)
+	}
+
+	if options.Flatten {
+		for _, suffix := range options.FlattenSuffixes {
+			secretName, err := k.makeSecretName(secrets.SecretKey{Name: key.Name + "-" + suffix, Namespace: key.Namespace, Kind: key.Kind})
+			if err != nil {
+				return errors.Wrapf(err, "couldn't make secret name for %+v", key)
```

Comment:
Don't we run a PII disclosure risk using `%+v` here with a secret?

Reply:
As above, fixed this